ui/components/app/snaps/snap-permission-cell-display/snap-permission-cell-display.js

```diff
@@ -0,0 +1,121 @@
+import PropTypes from 'prop-types';
+import { SubjectType } from '@metamask/permission-controller';
+import React from 'react';
+import {
+  AvatarBase,
+  AvatarBaseSize,
+  AvatarFavicon,
+  AvatarFaviconSize,
+  IconSize,
+  Text,
+} from '../../../component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  Display,
+  FontWeight,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import { getAvatarFallbackLetter } from '../../../../helpers/utils/util';
+import PermissionCell from '../../permission-cell';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+export default function SnapPermissionCellDisplay({
+  snapId,
+  showOptions,
+  permission,
+  subjectMetadata,
```

Comment:
We should document that this is the subject metadata of the target of the permission. Maybe even rename 🤔

Reply:
Renamed, should be more clear now.